extensions/amp-access-laterpay/0.1/laterpay-impl.js

```diff
@@ -437,7 +439,7 @@ export class LaterpayVendor {
     a.href = href;
     a.textContent = this.i18n_.alreadyPurchasedLink;
     this.alreadyPurchasedListener_ = listen(a, 'click', ev => {
-      this.handlePurchase_(ev, href);
+      this.handlePurchase_(ev, href, 'alreadyPurchased');
```

Comment:
I think that change would actually not be as simple as the one I proposed here, as the `handlePurchase_` method was getting the dataset out of `selectedPurchaseOption_` which is one of the radio buttons.

I could set that `selectedPurchaseOption_` property to the anchor instead, in the moment the user clicks the anchor, but then I'd also need to reset it in case the login flow fails and the user comes back to the page, to then try and select a different radio option (which could definitely happen in a case where the user thinks they've already purchased an article but actually haven't).

Overall that I feel like that would be more difficult to understand in terms of the overall flow of the code.